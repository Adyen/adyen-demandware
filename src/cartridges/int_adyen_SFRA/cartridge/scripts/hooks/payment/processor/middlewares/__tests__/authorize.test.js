/* eslint-disable global-require */
let authorize;
let currentBasket;

beforeEach(() => {
  authorize = require('../authorize');
  jest.clearAllMocks();
  const { getCurrentBasket } = require('dw/order/BasketMgr');
  currentBasket = getCurrentBasket();
});

afterEach(() => {
  jest.resetModules();
});

describe('adyen component authorize function', () => {
  it('return with appropriate mesages when create payment request fails', () => {
    const { createPaymentRequest } = require('*/cartridge/scripts/adyenCheckout');
    createPaymentRequest.mockImplementation(() => ({
      error: {},
    }));
    const authorizeResult = authorize(
      '15',
      currentBasket.createPaymentInstrument(),
      'mockedPaymentProcessor',
    );
    expect(authorizeResult).toMatchSnapshot();
  });

  it('handle the create payment request result 3DS', () => {
    const { createPaymentRequest } = require('*/cartridge/scripts/adyenCheckout');
    createPaymentRequest.mockImplementation(() => ({
      resultCode: 'RedirectShopper',
      redirectObject: {
        data: {
          MD: 'mockedMD',
        },
      },
    }));
    const authorizeResult = authorize(
      '15',
      currentBasket.createPaymentInstrument(),
      'mockedPaymentProcessor',
    );
    expect(authorizeResult).toMatchSnapshot();
  });

  it('handle the create payment request result 3DS2', () => {
    const { createPaymentRequest } = require('*/cartridge/scripts/adyenCheckout');
    createPaymentRequest.mockImplementation(() => ({
      threeDS2: 'mockedthreeDS2',
      resultCode: 'mockedresultCode',
      token3ds2: 'mockedtoken3ds2',
    }));
    const authorizeResult = authorize(
      '15',
      currentBasket.createPaymentInstrument(),
      'mockedPaymentProcessor',
    );
    expect(authorizeResult).toMatchSnapshot();
  });

  it('handle the create payment request result redirectShopper', () => {
    const { createPaymentRequest } = require('*/cartridge/scripts/adyenCheckout');
    createPaymentRequest.mockImplementation(() => ({
      resultCode: 'RedirectShopper',
      redirectObject: {
        url: 'mockedURL',
      },
      paymentData: 'mockedpaymentData',
    }));
    const authorizeResult = authorize(
      '15',
      currentBasket.createPaymentInstrument(),
      'mockedPaymentProcessor',
    );
    expect(authorizeResult).toMatchSnapshot();
  });

  it('handle the create payment request decision accept', () => {
    const { createPaymentRequest } = require('*/cartridge/scripts/adyenCheckout');
    createPaymentRequest.mockImplementation(() => ({
      decision: 'ACCEPT',
    }));
    const authorizeResult = authorize(
      '15',
      currentBasket.createPaymentInstrument(),
      'mockedPaymentProcessor',
    );
    expect(authorizeResult).toMatchSnapshot();
  });

  it('handle create payment request decisions other than accept', () => {
    const { createPaymentRequest } = require('*/cartridge/scripts/adyenCheckout');
    createPaymentRequest.mockImplementation(() => ({
      decision: "DON'T ACCEPT",
    }));
    const authorizeResult = authorize(
      '15',
      currentBasket.createPaymentInstrument(),
      'mockedPaymentProcessor',
    );
    expect(authorizeResult).toMatchSnapshot();
  });
});
