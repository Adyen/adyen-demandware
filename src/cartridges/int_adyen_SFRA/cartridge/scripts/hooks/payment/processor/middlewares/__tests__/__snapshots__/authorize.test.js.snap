// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adyen component authorize function handle create payment request decisions other than accept 1`] = `
Object {
  "error": true,
}
`;

exports[`adyen component authorize function handle the create payment request decision accept 1`] = `
Object {
  "authorized": true,
  "error": false,
}
`;

exports[`adyen component authorize function handle the create payment request result 3DS 1`] = `
Object {
  "authorized": true,
  "authorized3d": true,
  "orderNo": "15",
  "paymentInstrument": Object {
    "custom": Object {
      "adyenPaymentData": undefined,
    },
    "paymentTransaction": Object {
      "paymentProcessor": "mockedPaymentProcessor",
    },
    "setCreditCardExpirationMonth": [MockFunction],
    "setCreditCardExpirationYear": [MockFunction],
    "setCreditCardNumber": [MockFunction],
    "setCreditCardToken": [MockFunction],
    "setCreditCardType": [MockFunction],
  },
  "redirectObject": Object {
    "data": Object {
      "MD": "mockedMD",
    },
  },
  "signature": null,
}
`;

exports[`adyen component authorize function handle the create payment request result 3DS2 1`] = `
Object {
  "resultCode": "mockedresultCode",
  "threeDS2": "mockedthreeDS2",
  "token3ds2": "mockedtoken3ds2",
}
`;

exports[`adyen component authorize function handle the create payment request result redirectShopper 1`] = `
Object {
  "authorized": true,
  "authorized3d": false,
  "orderNo": "15",
  "paymentInstrument": Object {
    "custom": Object {
      "adyenPaymentData": "mockedpaymentData",
    },
    "paymentTransaction": Object {
      "paymentProcessor": "mockedPaymentProcessor",
    },
    "setCreditCardExpirationMonth": [MockFunction],
    "setCreditCardExpirationYear": [MockFunction],
    "setCreditCardNumber": [MockFunction],
    "setCreditCardToken": [MockFunction],
    "setCreditCardType": [MockFunction],
  },
  "redirectObject": Object {
    "url": "mockedURL",
  },
  "signature": "mockedURL",
}
`;

exports[`adyen component authorize function return with appropriate mesages when create payment request fails 1`] = `
Object {
  "authorized": false,
  "error": true,
  "fieldErrors": Array [],
  "serverErrors": Array [
    "mocked_error.payment.processor.not.supported",
  ],
}
`;
